rasa/shared/core/training_data/story_reader/yaml_story_reader.py

```diff
@@ -175,11 +176,18 @@ def is_key_in_yaml(cls, file_path: Union[Text, Path], *keys: Text) -> bool:
               `True` if all the keys are contained in the file, `False` otherwise.
 
         Raises:
-            YamlException: if the file seems to be a YAML file (extension) but
-                can not be read / parsed.
+            FileNotFoundException: if the file cannot be found.
         """
-        content = rasa.shared.utils.io.read_yaml_file(file_path)
-        return any(key in content for key in keys)
+        try:
+            with open(file_path) as file:
+                return all(
+                    any(line.startswith(f"{key}:") for line in file) for key in keys
```

Comment:
Optional: would be faster to do one pass of the lines and only check remaining keys.
Also if we change the behaviour back to `any`, then it would be much faster.

Reply:
Updated